### PR TITLE
Size task

### PR DESCRIPTION
**Pull request by clydin**

### CODE REVIEW

File: functions/src/plugins/size.ts [hunk 1]
```diff
@@ -110,50 +146,47 @@ export class SizeTask extends Task {
...
-              }
-              if(!lastNestedItemRef[encodedPath]) {
-                lastNestedItemRef[encodedPath] = {};
-              }
-
-              lastNestedItemRef = lastNestedItemRef[encodedPath];
+  async upsertNewArtifacts(
+    context: Context,
+    artifacts: BuildArtifact[]
+  ): Promise<void> {
```
Comment (ocombe) [inline functions/src/plugins/size.ts]: Can you put the function definition on one line instead of 4 ?

**Comment by ocombe** [comment]
Seems good, just some minor cosmetic remarks

File: functions/src/plugins/size.ts [hunk 2]
```diff
@@ -18,89 +15,128 @@ export interface CircleCiArtifact {
...
+        `Waiting for "${config.circleCiStatusName}"...`,
+        config.status.context,
+        context,
+      );
+
+      return;
+    } else if (statusEvent.state === STATUS_STATE.Failure) {
+      await this.setStatus(
+        STATUS_STATE.Error,
+        `Unable to calculate sizes.  Failure: "${config.circleCiStatusName}"`,
```
[the remark this answers is not in the document]
Comment (clydin) [reply functions/src/plugins/size.ts]: Done.

File: functions/src/plugins/size.ts (hunk 1 above)
Comment (clydin) [reply functions/src/plugins/size.ts]: Done.

**Pull request merged**